dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx

```diff
@@ -1265,6 +1270,123 @@ export function JsonRpcContextProvider({
     ),
   };
 
+  // -------- XRPL RPC METHODS --------
+
+  const xrplRpc = {
+    testSignTransaction: _createJsonRpcRequestHandler(
+      async (
+        chainId: string,
+        address: string
+      ): Promise<IFormattedRpcResponse> => {
+        try {
+          const result = await client!.request<{
+            tx_json: Record<string, any>;
+          }>({
+            chainId,
+            topic: session!.topic,
+            request: {
+              method: DEFAULT_XRPL_METHODS.XRPL_SIGN_TRANSACTION,
+              params: {
+                tx_json: {
+                  TransactionType: "OfferCreate",
+                  Account: address,
+                  Flags: 524288,
+                  TakerGets: "15000000000",
+                  TakerPays: {
+                    currency: "USD",
+                    issuer: "rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59B",
+                    value: "7072.8",
+                  },
+                },
```

Comment:
Could we consider using a simpler TX e.g. a regular Payment. 
```javascript
              params: {
                tx_json: {
                  Account: address,
                  Destination: "rTooLkitCksh5mQa67eaa2JaWHDBnHkpy",
                  Amount: "1",
                  TransactionType: "Payment",
                },
                submit: true,
                autofill: true,
              },
```

Also keep the amounts to a minimum value to not drain a wallet.

Reply:
Fixed for `xrpl_signTransaction`.
How do you feel about `xrpl_signTransaction_for`?